docs/src/architecture/accounts.md

```diff
@@ -1,27 +1,45 @@
 # Accounts
-An account is an entity which holds assets and defines rules of how these assets can be transferred. The diagram below illustrates basic components of an account.
+An account is an entity which holds assets and defines rules of how these assets can be transferred. The diagram below illustrates basic components of an account. In Miden every Account is a smart contract.
 
 <p align="center">
-    <img src="../diagrams/protocol/account/Account_Definition.png">
+    <img src="../diagrams/architecture/account/Account_Definition.png">
 </p>
 
 In the above:
 
-* ID is a unique identifier (32 bytes) of an account which does not change throughout its lifetime. This is not the same as address as accounts do not have addresses.
-* Storage is user-defined data which can be stored in an account. The exact mechanism is TBD. For example, this could be a key-value map or an index-based array.
-* Vault is collection of assets stored in an account. In this collection all fungible assets with the same ID are grouped together, and all non-fungible assets are represented as individual entries.
-* Code is a collection of functions which define an external interface for an account.
+* **Account ID** is a unique identifier (8 bytes) of an account which does not change throughout its lifetime. Account ID consists of 1 [field element](https://0xpolygonmiden.github.io/miden-vm/intro/overview.html) (~64 bits). This field element uniquely identifies a single account and also specifies the type of the underlying account. 
+* **Storage** is user-defined data which can be stored in an account. The exact mechanism is TBD. For example, this could be a key-value map or an index-based array.
+* **Nonce** is the account's nonce to be used when it creates a note. 
+* **Vault** is collection of assets stored in an account. In this collection all fungible assets with the same ID are grouped together, and all non-fungible assets are represented as individual entries.
+* **Code** is a collection of functions which define an external interface for an account.
```

Comment:
I would add here that the internal state of the account can be modified only by invoking these functions.